### ISSUE TITLE
Consider adding native Scintilla Autocomplete

### ISSUE THREAD

**Issue by vinsworldcom** [pnedev/nppgtags#41]
Would you consider adding an option (perhaps a checkbox in settings) to use Scintilla autocomplete instead of a plugin "native" version?  This would mean that simply typing would also trigger autocompletes (so long as enabled in Notepad++) for tag database items.

Scintilla autocomplete has a lot to be desired and there is not way to integrate with the already existing Notepad++ autocomplete features, but with Scintilla autocomplete, it would "just happen" when typing vs. having to setup yet another shortcut key for autocompletion.

I do something similar in (my fork of) [TagLeet](https://github.com/vinsworldcom/nppTagLEET).

Cheers.

**Comment by pnedev** [comment]
Hi again Vince,

Could you please check this development build: [**64-bit**](https://ci.appveyor.com/api/buildjobs/3vd65r35so1k7ouv/artifacts/NppGTags.x64.Release.v143.dll), [**32-bit**](https://ci.appveyor.com/api/buildjobs/m4gtap3t47w1omb2/artifacts/NppGTags.x86.Release.v143.dll). It is using the native plugin autocomplete list now but it is auto-shown on char input (after the third entered character). Is this example kind-of what you are expecting and proposing?

P.S. I see you have made an example PR but I don't have time to look at it right now. Sorry about that and thanks :+1:

**Comment by vinsworldcom** [comment]
Some testing and this simple change removes the annoying beep for me:

nppgtags/src/AutoCompleteWin.cpp:
```c++
448:                 case LVN_KEYDOWN:
449:                     if (ACW->onKeyDown(((LPNMLVKEYDOWN)lParam)->wVKey))
450:-                         return 0;
450:+                         return 1;
451:                 break;
```


Cheers.

**Comment by pnedev** [comment]
Thanks Vince for debugging the beep issue, the feedback and the testing :+1: 
You are right, I haven't implemented the NppGTags autocomplete function with multi-caret in mind.

BR

**Comment by vinsworldcom** [comment]
Yes, I use multi-caret quite regularly.  It would be unfortunate if it could not work in multi-caret mode.  The Scintilla native autocomplete does work in multi-caret mode.

I think I've corrected most of the problems with my PR.  I'm using it now in my Notepad++ daily use to see if any other issues come up.

Cheers.

**Comment by pnedev** [comment]
OK, I'll try it myself soon and probably merge it.
Thank you for looking into this.

P.S. Later I'll also add automatic autocomplete on char input + starting character per database setting.
You might also want to check my latest commits to dev branch - I have done some optimizations and fixes (including what you have already found about the "Create missing database" prompt on automatic autocomplete). One of those is a single check per file activation if it is included in a database and "caching" the result so no extensive database path lookup is done on each char input while editing a single file. There are some more fixes that might not be relevant to your implementation with Scintilla autocomplete.

**Comment by pnedev** [comment]
@vinsworldcom ,
Hello Vince,

I have added a setting to make Autocomplete automatically triggered after entering word with configurable length.
It is not native Scintilla autocomplete though (I prefer it that way at the moment) but it manages multi-selections without problem.
It will be available in the next official release but you can get the dev build from these links: [x64](https://ci.appveyor.com/api/buildjobs/ec5ltiu9mqdbuuwa/artifacts/NppGTags.x64.Release.v143.dll) / [x86](https://ci.appveyor.com/api/buildjobs/hli887t7g0n0wls6/artifacts/NppGTags.x86.Release.v143.dll).

I am sorry that it will require you to rework your patch to be able to sync to the latest official state.

BR
